src/statistics.jl

```diff
@@ -1,9 +1,9 @@
-using StatsBase: AbstractWeights, Weights, ProbabilityWeights, values, varcorrection, covm
+using StatsBase: AbstractWeights, ProbabilityWeights, values, varcorrection
 
 _unit_weights(n::Int) = ProbabilityWeights(ones(n), n)
 
 @doc doc"""
-    GeodesicInterpolationMethod <: AbstractMethod
+    GeodesicInterpolation <: AbstractEstimationMethod
```

Comment:
To some extend both GradientDescent and CyclicProximalPoint are also estimation Methods in this context (and it would avoid a conflict with Manopt that also will have an AbstractOptimizationMethod ;)). What do you think?

Reply:
This was from a suggestion by @mateuszbaran . I guess it depends on whether you will want `GradientDescent` and `CyclicProximalPoint` for anything else in Manifolds besides for these stats functions.

Reply:
Indeed that depends. My reasoning is, that I don't see that we will, so in that sense they are Estimatormethods at least here, I think?

Reply:
If both are supposed to be used as estimation methods only, they should inherit from `AbstractEstimationMethod`, and `GradientDescent` should be renamed to `GradientDescentEstimation` or something like that. Ultimately I think both `AbstractEstimationMethod` and `AbstractOptimizationMethod` should be added to `ManifoldsBase.jl`.

Reply:
For now, I'll make them both `AbstractEstimationMethod` and move their definitions to stats, since estimators should only be used in that context. What's the procedure for getting functionality into ManifoldsBase but still having it here for testing? I recommend we adopt a continuous delivery approach.

Reply:
In the long run `AbstractEstimationMethod` should go to `ManifoldsBase.jl`, yes, and maybe `AbstractOptimizationMethod`, too – though I'd happily "host" that in `Manopt`, too :)

Reply:
> What's the procedure for getting functionality into ManifoldsBase but still having it here for testing? I recommend we adopt a continuous delivery approach.

I think it should look like this:
1. Make changes to `ManifoldsBase.jl`.
2. Release a new version of `ManifoldsBase.jl`.
3. Use the new version of `ManifoldsBase.jl` in a PR to `Manifolds.jl`, lower-bounding the dependency.

Is the continuous delivery approach different?

Reply:
I think that's a good way. And we can do that whenever we decide that some feature (function, type, thtt,...) should go to `ManifoldsBase.jl`.

Reply:
No, that's basically continuous delivery, with the added twist that in the CD paradigm, you usually tag a new release after every PR, though I sometimes wait if I plan on finishing 2 PRs that go together back-to-back. See e.g. https://white.ucc.asn.au/2019/09/28/Continuous-Delivery-For-Julia-Packages.html

Reply:
From a certain status onwards, I would favour this mode (maybe for `ManifoldsBase.jl` already), our package here still has too many discussions (and not yet “foreign” PRs). But from a first public version onwards – yes, why not.